WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetListViewModel.kt

```diff
@@ -0,0 +1,106 @@
+@file:JvmName("StatsViewsWidgetConfigureViewModelKt")
+
+package org.wordpress.android.ui.stats.refresh.lists.widget
+
+import androidx.annotation.LayoutRes
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.stats.LimitMode.Top
+import org.wordpress.android.fluxc.model.stats.time.VisitsAndViewsModel.PeriodData
+import org.wordpress.android.fluxc.network.utils.StatsGranularity.DAYS
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.stats.time.VisitsAndViewsStore
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ValueItem.State.NEGATIVE
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ValueItem.State.NEUTRAL
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ValueItem.State.POSITIVE
+import org.wordpress.android.ui.stats.refresh.lists.sections.granular.usecases.OverviewMapper
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsViewsWidgetConfigureViewModel.Color
+import org.wordpress.android.ui.stats.refresh.utils.StatsDateFormatter
+import org.wordpress.android.viewmodel.ResourceProvider
+import java.util.Date
+import javax.inject.Inject
+
+const val LIST_ITEM_COUNT = 7
+
+class ViewsListViewModel
+@Inject constructor(
+    private val siteStore: SiteStore,
+    private val visitsAndViewsStore: VisitsAndViewsStore,
+    private val overviewMapper: OverviewMapper,
+    private val resourceProvider: ResourceProvider,
+    private val statsDateFormatter: StatsDateFormatter
+) {
+    private var siteId: Long? = null
+    private var colorModeId: Int? = null
+    private var showChangeColumn: Boolean = true
+    private val mutableData = mutableListOf<ListItemUiModel>()
+    val data: List<ListItemUiModel> = mutableData
+    fun start(siteId: Long, colorModeId: Int, showChangeColumn: Boolean) {
+        this.siteId = siteId
+        this.colorModeId = colorModeId
+        this.showChangeColumn = showChangeColumn
+    }
+
+    fun onDataSetChanged() {
+        siteId?.let {
+            val site = siteStore.getSiteBySiteId(it)
+            runBlocking {
+                visitsAndViewsStore.fetchVisits(site, DAYS, Top(LIST_ITEM_COUNT + 1), Date())
```

Comment:
If we delete the site used in widget from the app, `site` will be `null`, and app will crash around here.